sdk/tables/azure-data-tables/tests/test_table_batch.py

```diff
@@ -557,111 +611,129 @@ def test_batch_reuse(self, resource_group, location, storage_account, storage_ac
+            batch.create_entity(entity)
 
-            self.table.commit_batch(batch)
-            table2.commit_batch(batch)
+            self.table.send_batch(batch)
+            table2.send_batch(batch)
 
             batch = TableBatchClient()
             entity.RowKey = 'batch_all_operations_together'
-            batch.create_item(entity)
+            batch.create_entity(entity)
             entity.RowKey = 'batch_all_operations_together-1'
-            batch.delete_item(entity.PartitionKey, entity.RowKey)
+            batch.delete_entity(entity.PartitionKey, entity.RowKey)
             entity.RowKey = 'batch_all_operations_together-2'
             entity.test3 = 10
-            batch.update_item(entity)
+            batch.update_entity(entity)
             entity.RowKey = 'batch_all_operations_together-3'
             entity.test3 = 100
-            batch.update_item(entity, mode='MERGE')
+            batch.update_entity(entity, mode=UpdateMode.MERGE)
             entity.RowKey = 'batch_all_operations_together-4'
             entity.test3 = 10
-            batch.upsert_item(entity)
+            batch.upsert_entity(entity)
             entity.RowKey = 'batch_all_operations_together-5'
-            batch.upsert_item(entity, mode='MERGE')
+            batch.upsert_entity(entity, mode=UpdateMode.MERGE)
 
-            self.table.commit_batch(batch)
-            resp = table2.commit_batch(batch)
+            self.table.send_batch(batch)
+            resp = table2.send_batch(batch)
 
             # Assert
             self.assertEqual(6, len(list(resp)))
-            entities = list(self.table.query_items("PartitionKey eq '003'"))
+            entities = list(self.table.query_entities("PartitionKey eq '003'"))
             self.assertEqual(5, len(entities))
         finally:
             self._tear_down()
 
-    @pytest.mark.skip("pending")
+    # @pytest.mark.skip("This does not throw an error, but it should")
     @GlobalStorageAccountPreparer()
     def test_batch_same_row_operations_fail(self, resource_group, location, storage_account, storage_account_key):
         # Arrange
         self._set_up(storage_account, storage_account_key)
         try:
             entity = self._create_random_entity_dict('001', 'batch_negative_1')
-            self.table.create_item(entity)
+            self.table.create_entity(entity)
 
             # Act
             batch = self.table.create_batch()
 
             entity = self._create_updated_entity_dict(
                 '001', 'batch_negative_1')
-            batch.update_item(entity)
+            batch.update_entity(entity)
             entity = self._create_random_entity_dict(
                 '001', 'batch_negative_1')
-
+            batch.update_entity(entity, mode=UpdateMode.MERGE)
             # Assert
-            with self.assertRaises(ValueError):
-                batch.update_item(entity, mode='MERGE')
+            with self.assertRaises(HttpResponseError):
```

Comment:
We should be catching the BatchErrorException

Reply:
To clarify - where we get a non-200/201 status code, that should be an HttpResponseError.
Where we get a 200/201 status code - but individual responses within the batch had failures - that should be a BatchErrorException